scrapy/downloadermiddlewares/redirect.py

```diff
@@ -55,28 +55,25 @@ class RedirectMiddleware(BaseRedirectMiddleware):
 
     def process_response(self, request, response, spider):
         if (request.meta.get('dont_redirect', False) or
-               response.status in getattr(spider, 'handle_httpstatus_list', []) or
-               response.status in request.meta.get('handle_httpstatus_list', []) or
-               request.meta.get('handle_httpstatus_all', False)):
+                response.status in getattr(spider, 'handle_httpstatus_list', []) or
+                response.status in request.meta.get('handle_httpstatus_list', []) or
+                request.meta.get('handle_httpstatus_all', False)):
             return response
 
-        if request.method == 'HEAD':
-            if response.status in [301, 302, 303, 307] and 'Location' in response.headers:
-                redirected_url = urljoin(request.url, response.headers['location'])
+        location = None
+        if 'Location' in response.headers:
+            location = response.headers['location'].decode('latin1')
+
+        if location is not None and response.status in [301, 302, 303, 307]:
+            redirected_url = urljoin(request.url, location)
```

Comment:
In Python 2 we want URLs to be bytes; non-ascii argument to urljoin promotes `redirected_url` to unicode, and because of that `Request._set_url` will encode it to bytes using utf8; then it is percent-encoded.

So if Location header was Latin1, request URL will be sent as percent-escaped UTF8. Is it correct? If Location header was UTF8 it is even worse, we'll create garbage by decoding from Latin1 and then encoding this garbage to UTF8.

So the question is what browsers send exactly, assuming Location is in Latin1? Percent-encoded Latin1? We should send the same in Python 2.x and 3.x.